### DISCUSSION TITLE
album_art function

### DISCUSSION THREAD

**Discussion by Bruno21** [avantrec/soco-cli#16] [Q&A] [answered]
Hello,

Could you provide an example with the album_art function ?

I try from local file:

sonos -l Salon play_file 04.\ Delenda.mp3

from library:
sonos -l Salon queue_search_result_number 2 first : Salon play_from_queue

from Sono app

Always same result:

❯ sonos -l Salon album_art
Album art not accessible

**Comment by pwt** [comment, marked as answer]
`album_art` will only return a value if there's a usable HTTP URL available for the art. Some services provide this -- e.g., if playing a track via AirPlay 2, or from Sonos Radio.

Many sources do not, however. E.g., playing from a local library just presents a CIFS link, which isn't easy to use. I could just return this link, regardless of whether it's useable or not. Let me know if you'd prefer that.

(Note that I have no plans to do anything more sophisticated like querying external music databases.)

**Reply by Bruno21** [reply]
I misunderstood the purpose of album_art function. 
I thought I would use it to display the image of ID3 tags

Thanks for the precision.

**Comment by pwt** [comment]
Just to note that v0.4.5 will improve the `album_art` action to return a URL to the album art image in many more circumstances.
